### PR TITLE
feat(project-creation): Add framework selection modal

### LINKED ISSUE #48869: Framework selection modal in new project creation
The framework selection modal showed very good results in the new customer onboarding flow, so we should also add it to the new project creation.

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Preview:**

https://github.com/getsentry/sentry/assets/29228205/37e0fb66-4231-4de4-aecb-f06f5e37d2ec

closes:  https://github.com/getsentry/sentry/issues/48869

**will be done in a follow-up:**

- ~Unit Tests~ -> https://github.com/getsentry/sentry/pull/49305

### CODE REVIEW

File: static/app/views/projectInstall/createProject.spec.jsx [hunk 1]
```diff
@@ -67,42 +67,6 @@ describe('CreateProject', function () {
     expect(wrapper.container).toSnapshot();
   });
 
-  it('should fill in platform name if its provided by url', function () {
```
Comment (priscilawebdev) [inline static/app/views/projectInstall/createProject.spec.jsx:70]: I couldn't find a case where this happens.

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 2]
```diff
@@ -76,33 +78,48 @@ export function FrameworkSuggestionModal({
   );
 
   useEffect(() => {
-    trackAnalytics('onboarding.select_framework_modal_rendered', {
-      platform: selectedPlatform.key,
-      organization,
-    });
-  }, [selectedPlatform.key, organization]);
+    if (newOrg) {
```
Comment (matejminar) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:81]: Should we also track this for orgs that are not new?
Either under a different tracking key or the same one with different parameter.

File: static/app/views/projectInstall/overview.tsx [hunk 3]
```diff
@@ -56,12 +59,18 @@ class ProjectInstallOverview extends AsyncComponent<Props, State> {
...
 
   render() {
-    const {organization} = this.props;
+    const {organization, projects} = this.props;
     const {projectId} = this.props.params;
     const {keyList, showDsn} = this.state;
 
     const issueStreamLink = `/organizations/${organization.slug}/issues/#welcome`;
 
+    const project = projects.find(p => p.slug === projectId);
```
Comment (matejminar) [inline static/app/views/projectInstall/overview.tsx:68]: Does `withProjects` guarantee that all projects will be returned? Even if org has hundreds of projects?

**Review by matejminar — APPROVED** [review]

File: static/app/components/onboarding/frameworkSuggestionModal.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/components/onboarding/frameworkSuggestionModal.tsx:81]: yes I plan to do that in a separate PR

File: static/app/views/projectInstall/overview.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/views/projectInstall/overview.tsx:68]: I didn't want to convert this component just now to be able to use `useProjects` but this is done in a follow-up :) https://github.com/getsentry/sentry/pull/49303

**Pull request merged**